bin/extract_codeobject

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/perl
+use strict;
+use File::Copy;
+use File::Spec;
+use File::Basename;
+use File::Which;
+use Cwd 'realpath';
+use Getopt::Std;
+use List::Util qw(max);
+
+sub usage {
+  print("Usage: $0 [OPTION]... -i <input>\n");
+  print("Extract the device kernels from an hcc executable.\n\n");
+  print("-h                                \tshow this help message\n");
+  print("-i <input>                        \t[DEPRECATED] input file\n");
```

Comment:
Why is this deprecated? Seems it is needed for the -l option.

Would -e be a better name for the host executable?

-e <host-executable> \tSpecify host executable file.